nixos/modules/programs/vivid.nix

```diff
@@ -0,0 +1,63 @@
+{
+  pkgs,
+  config,
+  lib,
+  ...
+}:
+
+let
+  cfg = config.programs.vivid;
+in
+{
+  options = {
+    programs.vivid = {
+      enable = lib.mkOption {
+        default = false;
+        description = "Whether to configure LS_COLORS with vivid.";
+        type = lib.types.bool;
+      };
+
+      package = lib.mkPackageOption pkgs "vivid" { example = "vivid"; };
+
+      theme = lib.mkOption {
+        default = "gruvbox-dark-soft";
+        description = "Theme to be used (see `vivid themes`)";
+        example = "solarized-dark";
+        type = lib.types.str;
+      };
+    };
+  };
+
+  config = lib.mkIf cfg.enable {
+    environment.systemPackages = [ cfg.package ];
+
+    programs =
+      let
+        interactiveShellInit = lib.mkAfter ''
+          export LS_COLORS="$(${lib.getExe cfg.package} generate ${cfg.theme})"
+        '';
+        enableLsColors = lib.mkForce false;
```

Comment:
I would suggest a strength of default + 1, otherwise the assertion below doesn't really work (right?)

Reply:
> I would suggest a strength of default + 1,

I've used `mkOverride 999`, which is `mkDefault - 1` (so, a higher priority), is it correct or is there a more idiomatic way to do it?

> otherwise the assertion below doesn't really work (right?)

Let me come up with a simple case.

Reply:
> is it correct or is there a more idiomatic way to do it?

It seems like the best option to me. You essentially want to replace the default value with another default. There are 3 other uses of the pattern I can find.

Reply:
Assertions are working:

```
-- sandbox.nix
{ config, pkgs, ... }:
{
  boot.loader.systemd-boot.enable = true;
  boot.loader.efi.canTouchEfiVariables = true;

  programs = {
    vivid.enable = true;
    bash.enableLsColors = true;
    zsh.enableLsColors = false;
  };

  system.stateVersion = "25.05";
}
```

gives:

```
$  nix-build './nixos' -A vm -I nixos-config=./sandbox.nix
error:
       … while calling the 'derivationStrict' builtin
         at <nix/derivation-internal.nix>:34:12:
           33|
           34|   strict = derivationStrict drvAttrs;
             |            ^
           35|

       … while evaluating derivation 'nixos-vm'
         whose name attribute is located at /home/black/nixpkgs/pkgs/stdenv/generic/make-derivation.nix:375:7

       … while evaluating attribute 'buildCommand' of derivation 'nixos-vm'
         at /home/black/nixpkgs/pkgs/build-support/trivial-builders/default.nix:59:17:
           58|         enableParallelBuilding = true;
           59|         inherit buildCommand name;
             |                 ^
           60|         passAsFile = [ "buildCommand" ]

       … while evaluating the option `virtualisation.vmVariant.system.build.toplevel':

       … while evaluating definitions from `/home/black/nixpkgs/nixos/modules/system/activation/top-level.nix':

       (stack trace truncated; use '--show-trace' to show the full, detailed trace)

       error:
       Failed assertions:
       - `programs.vivid.enable` is incompatible with `programs.bash.enableLsColors`.
```